scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala

```diff
@@ -337,6 +339,67 @@ object GetSettings extends EsScript {
     sys.exit(1)
   }
 }
+object DownloadAllEsIds extends EsScript {
+
+  def run(esUrl: String, extraArgs: List[String]) {
+
+    println(s"Getting all ids from $esUrl")
+
+    object Client extends EsClient(esUrl) {
+
+      def writeAllEsIdsToFile(filename: String) = {
+
+        val file = new File(filename)
+
+        val writer = new PrintWriter(file)
+
+        @tailrec
+        def processResponseAndStartNext(prevResponse: Response[SearchResponse]): Unit = {
+          if(prevResponse.status != 200) {
+            writer.close()
+            client.close()
+            throw new Exception("Failed performing search query")
+          }
+
+          println(s"Adding ${prevResponse.result.hits.hits.length} IDs to ${file.getAbsolutePath}")
+
+          prevResponse.result.hits.hits.map(_.id).foreach(writer.println)
+
+          // can't be map/foreach if we want tail recursion (for efficiency)
```

Comment:
but could be a match statement :)

Reply:
good suggestion, done in https://github.com/guardian/grid/pull/4114/commits/4fe250a12340b6ef41759080ced09dde6e6f5d9f and re-tested